### PR TITLE
(PUP-4351) Default service type and package provider not set on CumulusL...

### PR DESCRIPTION
**Pull request by LeslieCarr**
...inux

This fix adds CumulusLinux to the defaultfor for its package and service
provider.

Thanks also to Barry Peddycord (@isharacomix) for work on this patch

### CODE REVIEW

File: lib/puppet/provider/package/apt.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@
   commands :aptcache => "/usr/bin/apt-cache"
   commands :preseed => "/usr/bin/debconf-set-selections"
 
-  defaultfor :operatingsystem => [:debian, :ubuntu]
+  defaultfor :operatingsystem => [:debian, :ubuntu, :CumulusLinux]
```
Comment (haus) [inline lib/puppet/provider/package/apt.rb]: Any reason we can't just default on `:osfamily => [:debian]`?

Comment (LeslieCarr) [inline lib/puppet/provider/package/apt.rb]: I actually think that would be better but was a bit cautious in making a
more radical change. Should I submit an updated PR?

Comment (branan) [inline lib/puppet/provider/package/apt.rb]: @LeslieCarr yeah, would be great to see that.

**Comment by melissa** [comment]
@isharacomix since you're listed as a sub-contributor on this patch, could you also sign our CLA at http://cla.puppetlabs.com/? Thank you both for your work on this.

**Comment by LeslieCarr** [comment]
@haus how does this look now?

**Comment by haus** [comment]
:+1:

**Comment by branan** [comment]
@haus @LeslieCarr wait, is `osfamily` really `Debian` with a capital "D"? But lowercase when it's `operatingsystem`?

**Comment by haus** [comment]
```
matthaus@turtle:~$ facter osfamily
Debian
matthaus@turtle:~$ facter operatingsystem
Debian
```

**Comment by LeslieCarr** [comment]
I double checked `osfamily` is capitalized on ubuntu14.04, debian wheezy, and cumulus

**Comment by whopper** [comment]
To verify, in Facter 2.4, 'Debian' will be capitalized both in `osfamily` and `operatingsystem`:

     24       def get_osfamily
     ....
     28         when "LinuxMint", "Ubuntu", "Debian"
     29           "Debian"

    241       def get_operatingsystem_with_release_files
    ....
    245           "Debian"      => "/etc/debian_version",

File: lib/puppet/provider/service/debian.rb [hunk 2]
```diff
@@ -16,7 +16,7 @@
   # is resolved.
   commands :invoke_rc => "/usr/sbin/invoke-rc.d"
 
-  defaultfor :operatingsystem => :debian
+  defaultfor :operatingsystem => [:debian, :CumulusLinux]
```
Comment (melissa) [inline lib/puppet/provider/service/debian.rb]: @LeslieCarr could you capitalize `:debian` for the sake of readability?

**Comment by melissa** [comment]
@LeslieCarr we're just waiting on two more things before this can get updated.

First, since all calls to `defaultsfor` reference lowercase variables, both of the changes you make should reference lowercase variables as well. All those references are case insensitive, so even though facter returns capitalized variables, we should stick to lowercase references for consistency.

Second, if @isharacomix could sign the CLA as well, that would be wonderful. To do that, just go to http://cla.puppetlabs.com/ and follow the directions. Your agreement will be linked to your github account. 

Thank you again for your contribution! I'll keep checking up on this, but if you ping me when both of those changes are in, I can get this merged!

**Comment by melissa** [comment]
Also, to make sure that this question of capitalization doesn't bite us, I've written spec tests to make sure these default correctly on Cumulus Linux. That work is at https://github.com/puppetlabs/puppet/pull/3807

**Comment by LeslieCarr** [comment]
Everyone has now signed the CLA (should be covered under the blanket Cumulus Networks CLA) and sorry about the delay (yay lawyers!).  How's the update look?

**Comment by isharacomix** [comment]
Indeed, thanks @LeslieCarr for taking care of all of this. :)

**Pull request merged**